File: src/main/java/org/nest/nestml/_visitor/BooleanLiteralVisitor.java
package org.nest.nestml._visitor;

import org.nest.nestml._ast.ASTExpr;
import org.nest.nestml._symboltable.typechecking.Either;

import static org.nest.nestml._symboltable.predefined.PredefinedTypes.getBooleanType;


/**
 * @author ptraeder
 */
public class BooleanLiteralVisitor implements NESTMLVisitor {

  @Override
  public void visit(ASTExpr expr) {
    expr.setType(Either.value(getBooleanType()));
  }
}

File: pynestml/src/main/python/org/nestml/visitor/expression_visitor/BooleanLiteralVisitor.py
#
# BooleanLiteralVisitor.py
#
# This file is part of NEST.
#
# Copyright (C) 2004 The NEST Initiative
#
# NEST is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 2 of the License, or
# (at your option) any later version.
#
# NEST is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with NEST.  If not, see <http://www.gnu.org/licenses/>.

"""
simpleexpression : BOOLEAN_LITERAL // true & false;
"""
from pynestml.src.main.python.org.nestml.symbol_table.predefined.PredefinedTypes import PredefinedTypes
from pynestml.src.main.python.org.nestml.visitor.NESTMLVisitor import NESTMLVisitor
from pynestml.src.main.python.org.nestml.visitor.expression_visitor.Either import Either


class BooleanLiteralVisitor(NESTMLVisitor):

    def visitSimpleExpression(self, _expr=None):
        _expr.setTypeEither(Either.value(PredefinedTypes.getBooleanType()))
        return